photutils/psf/psfphotometry.py

```diff
@@ -0,0 +1,446 @@
+            `~astropy.modeling.fitting` for more details on fitters.
+        niters : int (default=3)
+            Number of iterations to perform the loop FIND, GROUP, SUBTRACT,
+            NSTAR.
+        aperture_radius : float (default=None)
+            The radius used to compute initial estimates for the fluxes of
+            sources. If ``None``, one fwhm will be used. 
+
+        Notes
+        -----
+        If there are problems with fitting large groups, change the parameters
+        of the grouping algorithm to reduce the number of sources in each
+        group or input a ``star_groups`` table that only includes the groups
+        that are relevant (e.g. manually remove all entries that coincide with
+        artifacts).
+
+        References
+        ----------
+        [1] Stetson, Astronomical Society of the Pacific, Publications,
+            (ISSN 0004-6280), vol. 99, March 1987, p. 191-222.
+            Available at: http://adsabs.harvard.edu/abs/1987PASP...99..191S
+        """
+
+        self.find = find
+        self.group = group
+        self.bkg = bkg
+        self.psf = psf
+        self.fitter = fitter
+        self.niters = niters
+        self.fitshape = fitshape
+        self.aperture_radius = aperture_radius
+
+        @property
+        def niters(self):
+            return self._niters
+
+        @niters.setter
+        def niters(self, niters):
+            if isinstance(niters, int) and niters > 0:
+                self._niters = niters
+            else:
+                raise ValueError('niters must be an interger-valued number, '
+                                 'received niters = {}'.format(niters))
+        
+        @property
+        def fitshape(self):
+            return self._fitshape
+
+        @fitshape.setter
+        def fitshape(self, fitshape):
+            fitshape = np.asarray(fitshape)
+            if len(fitshape) == 2:
+                if np.all(fitshape) > 0:
+                    if np.all(fitshape) % 2 == 1:
+                        self._fitshape = fitshape
+                    else:
+                        raise ValueError('fitshape must be odd '
+                                         'integer-valued, '
+                                         'received fitshape = {}'\
+                                         .format(fitshape))
+                else:
+                    raise ValueError('fitshape must have positive elements, '
+                                     'received fitshape = {}'\
+                                     .format(fitshape))
+            else:
+                raise ValueError('fitshape must have two dimensions, '
+                                 'received fitshape = {}'.format(fitshape))
+
+        @property
+        def aperture_radius(self):
+            return self._aperture_radius
+
+        @niters.setter
+        def aperture_radius(self, aperture_radius):
+            if isinstance(aperture_radius, (int, float)) and aperture_radius > 0:
```

Comment:
also, doesn't this prevent the "None" value from passing through?  That is, with this check in place, `None` can't be the value of `paerture_radius`?